src/utils/paths.cpp

```diff
@@ -68,6 +70,23 @@ std::filesystem::path Paths::getUserLockFilePath() {
 }
 
 std::filesystem::path Paths::getSystemConfigFilePath() {
+  char *xdgConfigDirs = getenv("XDG_CONFIG_DIRS");
+
+  if (xdgConfigDirs != nullptr) {
+    char *rest = nullptr;
```

Comment:
We should go the C++ way and take advantage of its templates. You can `#include "utils/string.h"` and use the `split` function defined there.

Then, since we are targeting C++ >= 11, you'll be able to easily iterate the vector using a range based for loop:
https://www.geeksforgeeks.org/range-based-loop-c/